### PR TITLE
DM-27321: Add day_obs and seq_num to exposure record

### PR DESCRIPTION
**Pull request by timj**
Also rename exposure.name to exposure.obs_id to make it look like obsCore.

### CODE REVIEW

File: python/lsst/daf/butler/configs/dimensions.yaml [hunk 1]
```diff
@@ -142,6 +142,12 @@ elements:
     requires: [instrument]
     implies: [physical_filter, visit_system]
     metadata:
+      -
+        name: day_obs
+        type: int
+        doc: >
+          Day of observation as defined by the observatory (YYYYMMDD format).
+          If a visit crosses multiple days this entry will be null.
```
Comment (RobertLuptonTheGood) [inline python/lsst/daf/butler/configs/dimensions.yaml]: Is this correct?  I thought it was the day when the observation is started;  LSST needs a dayObs for all data we take (not quite the same as this day_obs)

File: python/lsst/daf/butler/configs/dimensions.yaml [hunk 2]
```diff
@@ -227,6 +233,18 @@ elements:
...
+      -
+        name: day_obs
+        type: int
+        doc: >
+          Day of observation as defined by the observatory (YYYYMMDD format).
+      -
+        name: seq_num
+        type: int
+        doc: >
+          Counter for the observation within a larger sequence. Contxt
```
Comment (RobertLuptonTheGood) [inline python/lsst/daf/butler/configs/dimensions.yaml]: "Context"

**Review by RobertLuptonTheGood — APPROVED** [review]

File: python/lsst/daf/butler/configs/dimensions.yaml (hunk 1 above)
Comment (timj) [reply python/lsst/daf/butler/configs/dimensions.yaml]: I can change it for visit so that it returns the smallest day_obs if that's what you prefer.

Comment (RobertLuptonTheGood) [reply python/lsst/daf/butler/configs/dimensions.yaml]: Yes please.

**Pull request merged**